repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -197,18 +230,62 @@ object Evaluator{
       // Exhaust the printer iterator now, before exiting the `Catching`
       // block, so any exceptions thrown get properly caught and handled
       evaluatorRunPrinter(printer(evalMain(cls).asInstanceOf[Iterator[String]]))
+      evaluationResult(wrapperName, newImports)
+    }
+
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
```

Comment:
`processLine`'s algorithm is roughly compile, load, evaluate. `evalClass` does both the compiling and loading. `compileCacheBlock` compiles, then writes the cache, and then loads the class, and does no evaluation. (AFAIK the initializer runs only if the object is referenced somehow). The common code that made sense to me to pull is already in `loadClass` and `compileClass`.